### PR TITLE
notes

### PR DESCRIPTION
**Pull request by pdewilde**

### CODE REVIEW

File: pipeline/leech/main.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ func init() {
 	register.DoFn2x1[context.Context, leech.EventRecord, leech.LeechRecord](&leech.IngestLogsFn{})
 }
 
+// why is this file here and not in cmd/? is it because its not intended to be run from cli?
```
Comment (bradegler) [inline pipeline/leech/main.go:47]: Apache Beam pipelines aren't really executable in the same way as the rest of our commands. Putting it in `cmd` comes with an expectation that it is runnable without a beam runner as is which it is not.

File: pipeline/leech/main.go [hunk 2]
```diff
@@ -112,6 +113,8 @@ func realMain(ctx context.Context) error {
 	}
 
 	var event leech.EventRecord
+	// I know its not really the threat model, but it does seem to be a bad habit to use
+	// string replacement for query building.
```
Comment (bradegler) [inline pipeline/leech/main.go:117]: The go bigquery connector for Apache Beam doesn't support bind parameters so there isn't much choice unless we write our own BQ integration.

File: pkg/leech/ingest_logs.go [hunk 3]
```diff
@@ -124,10 +124,12 @@ func (f *IngestLogsFn) StartBundle(ctx context.Context) error {
 		return fmt.Errorf("failed to read private key: %w", err)
 	}
 	ghAppConfig := githubapp.NewConfig(f.GitHubAppID, f.GitHubInstallID, pk)
+	// Why not f.ghApp = githubapp.New(ghAppConfig)?
```
Comment (bradegler) [inline pkg/leech/ingest_logs.go:127]: Initial implementation didn't assign it to f.ghApp and when that change was made it wasn't cleaned up.

File: pkg/leech/ingest_logs.go [hunk 4]
```diff
@@ -124,10 +124,12 @@ func (f *IngestLogsFn) StartBundle(ctx context.Context) error {
...
 	ghApp := githubapp.New(ghAppConfig)
 	f.ghApp = ghApp
 
 	// setup the http client
+	// Why such a massive timeout? Just for large log files?
```
Comment (bradegler) [inline pkg/leech/ingest_logs.go:132]: Correct.

File: pkg/leech/ingest_logs.go [hunk 5]
```diff
@@ -138,6 +140,7 @@ func (f *IngestLogsFn) StartBundle(ctx context.Context) error {
 func (f *IngestLogsFn) ProcessElement(ctx context.Context, event EventRecord) LeechRecord {
 	logger := logging.FromContext(ctx)
 
+	// Why have this log as well as the processing element log a few lines down?
```
Comment (bradegler) [inline pkg/leech/ingest_logs.go:143]: It could be removed. Its a left over from early debugging.

File: pkg/leech/ingest_logs.go [hunk 6]
```diff
@@ -150,12 +153,15 @@ func (f *IngestLogsFn) ProcessElement(ctx context.Context, event EventRecord) Le
 		RepositoryName:   event.RepositoryName,
 		RepositorySlug:   event.RepositorySlug,
 		LogsURI:          gcsPath,
-		Status:           "SUCCESS",
+		Status:           "SUCCESS", // maybe initialize as PENDING and then update in/after handleMessage?
```
Comment (bradegler) [inline pkg/leech/ingest_logs.go:156]: In the majority of cases this should be `SUCCESS` but you certainly could do something else here.

File: pkg/leech/ingest_logs.go [hunk 7]
```diff
@@ -150,12 +153,15 @@ func (f *IngestLogsFn) ProcessElement(ctx context.Context, event EventRecord) Le
...
 	}
+	// Shouldn't this be Infow?
+	// it seems strange to log "result" when we mutate result.Status later on.
```
Comment (bradegler) [inline pkg/leech/ingest_logs.go:159]: Fair point, feel free to remove status.

File: pkg/leech/storage.go [hunk 8]
```diff
@@ -61,6 +61,7 @@ func (s *ObjectStore) Write(ctx context.Context, content io.Reader, objectDescri
 	writer := obj.NewWriter(ctx)
 
 	if _, err := io.Copy(writer, content); err != nil {
+		// Should we cancel context? obj.NewWriter() says to do so if you don't want ot write
```
Comment (bradegler) [inline pkg/leech/storage.go:64]: I don't see that in the docs, can you point me to a reference?

Comment (pdewilde) [reply pkg/leech/storage.go:64]: https://pkg.go.dev/cloud.google.com/go/storage#ObjectHandle.NewWriter

>It is the caller's responsibility to call Close when writing is done. To stop writing without saving the data, cancel the context.

Comment (pdewilde) [reply pkg/leech/storage.go:64]: Disappointingly, the example code neither closes nor cancels the context.

**Comment by sethvargo** [comment]
@pdewilde do we still need this PR? It looks like it was used as a way to ask questions about the code and we do not intend to merge it.

**Comment by pdewilde** [comment]
> @pdewilde do we still need this PR? It looks like it was used as a way to ask questions about the code and we do not intend to merge it.

Correct, I'll close it out

**Pull request closed without merging**